src/ImageSharp/Formats/Jpeg/Components/Decoder/SpectralConverter{TPixel}.cs

```diff
@@ -40,23 +43,20 @@ public SpectralConverter(Configuration configuration, CancellationToken cancella
 
         private bool Converted => this.pixelRowCounter >= this.pixelBuffer.Height;
 
-        public Buffer2D<TPixel> PixelBuffer
+        public Buffer2D<TPixel> GetPixelBuffer()
```

Comment:
There is heavy lazy initialization, with side effects like `CancellationToken` firing. It's better to expose such stuff from a method than a property.

Reply:
I didn't like this property approach with a lot of lazy stuff behind from the start but couldn't come up with anything better. There's a lot of confusing bits of code here and there from my pr, I will work on decoder performance/refactoring next as I'm done with the encoder for now.